js/components/Debugger.tsx

```diff
@@ -0,0 +1,143 @@
+import { h, JSX } from 'preact';
+import { useCallback, useEffect, useRef, useState } from 'preact/hooks';
+import { Apple2 as Apple2Impl } from '../apple2';
+import { Inset } from './Inset';
+
+import styles from './css/Debugger.module.css';
+import { ControlButton } from './ControlButton';
+
+export interface DebuggerProps {
+    apple2: Apple2Impl | undefined;
+    e: boolean;
+}
+
+interface DebugData {
+    memory: string;
+    registers: string;
+    running: boolean;
+    stack: string;
+    trace: string;
+    zeroPage: string;
+}
```

Comment:
It might be worth adding the `debug` output somewhere. Or, maybe, there should be another call to output status info to the debugger. This would make tracing through something like the RWTS code easier.

Reply:
That's something I hadn't thought of, but totally makes sense.